src/app/embed/adapters/merge.adapter.ts

```diff
@@ -0,0 +1,36 @@
+import {Injectable} from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { ReplaySubject } from 'rxjs';
+import { QSDAdapter } from './qsd.adapter'
+import { FeedAdapter } from './feed.adapter'
+import { AdapterProperties, hasMinimumParams, DataAdapter, getMergedValues } from './adapter.properties'
+
+const NO_EMIT_YET = Symbol();
+
+@Injectable()
+export class MergeAdapter {
+  private adapters: DataAdapter[]
+  constructor(
+    private FeedAdapter: FeedAdapter,
+    private QSDAdapter:  QSDAdapter
+	) { 
+    this.adapters = [this.QSDAdapter, this.FeedAdapter]
+  }
+
+  getProperties(params): Observable<AdapterProperties> {
+    let chosenAdapters: Observable<AdapterProperties>[];
+
+    chosenAdapters = this.adapters.
+      map(obs => obs.getProperties(params)).
+      map(obs => obs.startWith(NO_EMIT_YET))
+
+    return Observable.combineLatest(...chosenAdapters).map(sources => {
+      let data = [];
+      for (let source of sources) {
+        if (source === NO_EMIT_YET) { break; }
+        data.push(source);
+      }
```

Comment:
we walk through the results of the observables but stop as soon as we find one that hasn't emitted yet. That means that if there are 5 observables, and observables 1, 2, 4, and 5 have emitted, we only consider the values of 1 and 2, because the values of 4 and 5 *may* be overridden by the values in 3, which we have not seen yet.